src/main/webapp/static/js/service/pokemon_service.js

```diff
@@ -0,0 +1,77 @@
+'use strict';
+
+angular.module('myApp').factory('PokemonService', ['$http', '$q', function ($http, $q) {
```

Comment:
remove the $q dependency, and use promise syntax and returns instead to (See John Papa AngularJS 1 styleguide) to resolve the service calls.